### DISCUSSION TITLE
How to configure functional-parameters for React-Hooks

### DISCUSSION THREAD

**Discussion by 0815pascal** [jonaskello/eslint-plugin-functional#560] [Q&A] [answered]
Hi all, 

I was wondering how I need to configure the rule `functional/functional-parameters` so that it wouldn't be triggered in React-Hooks such as `useEffect` ? In my case I'm having

```javascript
useEffect(() => {  // Warning: Functions must have at least one parameter
...
}, [myVar]);
```

and .eslintrc is set to

```json
"functional/functional-parameters":  ["warn", { "ignorePattern": ["useEffect"]}],
```

I tried different `ignorePattern` but it seems like I'm doing something wrong. Or would I have to apply the `ignorePrefixSelector` ?

Thanks again!

**Comment by RebeccaStevens** [comment, marked as answer]
You would want to use the [`enforceParameterCount` option](https://github.com/eslint-functional/eslint-plugin-functional/blob/main/docs/rules/functional-parameters.md#enforceparametercount). You can simply just set this to `false`, or alternatively just set the [sub-option `ignoreLambdaExpression`](https://github.com/eslint-functional/eslint-plugin-functional/blob/main/docs/rules/functional-parameters.md#enforceparametercountignorelambdaexpression) to `true`.

```json
"functional/functional-parameters":  ["warn", { "enforceParameterCount": { "ignoreLambdaExpression": true } }],
```

**Reply by 0815pascal** [reply]
Thanks a lot! I have somehow misread the documentation and was thinking that these options were already active when extending the configuration with `"plugin:functional/recommended"` but now it does exactly what I was looking for.
